src/librustc/session/config.rs

```diff
@@ -268,22 +268,38 @@ impl OutputTypes {
 // DO NOT switch BTreeMap or BTreeSet out for an unsorted container type! That
 // would break dependency tracking for command-line arguments.
 #[derive(Clone, Hash)]
-pub struct Externs(BTreeMap<String, BTreeSet<Option<String>>>);
+pub struct Externs(BTreeMap<String, BTreeSet<ExternEntry>>);
+
+#[derive(Clone, Hash, Eq, PartialEq, Ord, PartialOrd, Debug)]
+pub struct ExternEntry {
+    pub location: Option<String>,
+    pub public: bool
+}
+
+impl ExternEntry {
+    pub fn new_public(location: Option<String>) -> ExternEntry {
```

Comment:
This function is only used from tests, could you move it into `mod tests` and make private.